pkgs/development/cuda-modules/generic-builders/manifest.nix

```diff
@@ -124,14 +125,22 @@ backendStdenv.mkDerivation (
       python = ["**/*.whl"];
     };
 
-    # Useful for introspecting why something went wrong.
-    # Maps descriptions of why the derivation would be marked broken to
-    # booleans indicating whether that description is true.
-    # NOTE: This should not include reasons relating to the architecture, as those are handled by
-    # the `badPlatforms` attribute.
+    # Useful for introspecting why something went wrong. Maps descriptions of why the derivation would be marked as
+    # broken on have badPlatforms include the current platform.
+
     # brokenConditions :: AttrSet Bool
+    # Sets `meta.broken = true` if any of the conditions are true.
+    # Example: Broken on a specific version of CUDA or when a dependency has a specific version.
     brokenConditions = { };
 
+    # badPlatformsConditions :: AttrSet Bool
+    # Sets `meta.badPlatforms = meta.platforms` if any of the conditions are true.
+    # Example: Broken on a specific architecture or when cudaSupport is false (building with CUDA essentially targets)
+    # a platform which NixOS doesn't have a notion of, otherwise we would specify the platform directly.
+    badPlatformsConditions = {
+      "CUDA support is disabled" = !config.cudaSupport;
```

Comment:
This is something new. What are the implications, what was the motivation?

Reply:
Since reverted -- the motivation was to see which packages were pulling in CUDA packages irrespective of whether `cudaSupport` was set to true and to investigate the impact of making cudaPackages unavailable when `cudaSupport` is false (by setting `badPlatforms`).

I'm still torn about such change because debugging is rough ("huh! Why are `platforms` and `badPlatforms` the same?") and I haven't found a way to issue a message to user that it's a result of depending on a member of `cudaPackages` without guarding against its use when `cudaSupport` is false.